### ISSUE TITLE
Fetch source from Github-provided URL instead of maintainter-provided upload

### ISSUE THREAD

**Issue by imchau** [flathub/org.keepassxc.KeePassXC#62]
The Flathub build is currently using the manually uploaded source tarballs from https://github.com/keepassxreboot/keepassxc/releases/ These are all manually uploaded by the listed author and could potentially contain different code than the publicly viewable repository on Github.

I don't know if and what verification process takes place on Flathub to ensure that the hosted applications are safe, so I would like to propose a minor change that is currently trivial to implement. I believe the following would increase the trustworthiness and security in all cases with minimal effort.

Github provides an automatic link to download the repository contents of a certain tag which are guaranteed to contain the publicly available repository.

In the case that one of the maintainers' accounts or even their goodwill were to be compromised, it would be practically impossible to hide malicious source code in the public repository of KeePassXC as it will be easily discovered and addressed. However, the prebuilt binaries and the source tarball currently used by Flathub from the releases page are uploaded by the maintainer and could theoretically be compiled from different sources. The source tarball could differ from the actual repository. 

For this reason, I think that the uploads provided by a project's maintainers are more prone to being compromised.

Therefore, I propose changing the URL for fetching the source from this:
`https://github.com/keepassxreboot/keepassxc/releases/download/2.6.0/keepassxc-2.6.0-src.tar.xz`
To this instead:
`https://github.com/keepassxreboot/keepassxc/archive/2.6.0.tar.gz`

The contents of both are nearly identical. The maintainer-uploaded release tarballs merely include two additional one-liner files over the Github-provided .tar.gz of the same tag.
Only in [.tar.xz]: .gitrev
Only in [.tar.xz]: .version

If these two files are necessary for a release build, they can be included in the Flatpak-specific patchset.

**Comment by AsavarTzeth** [comment]
I see. The main reason I use that source is that is what is linked to by the [upstream download page](https://keepassxc.org/download/#source) and it is signed by upstream developers (not that I can verify that automatically), but it is possible to do. It is also used by major distributions, like [Arch Linux](https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/keepassxc).

I'm not against this change per-say but I don't really see the security benefit. Surely if someone untrustworthy compromises the keepassxreboot github organization and is able to upload a different tarball, with a different signature (that the other core devs somehow overlook), they can do a lot more harm.

Could you also clarify:
- Who this "maintainer" is? The one that uploads the source tarball?`Surely that is a trusted person within the project?
- What pre compiled binaries are you referring to? I'm not using any binaries in this Flatpak.

**Comment by imchau** [comment]
I am referring to the case that the account or goodwill of one of the trusted persons within the project (maintainers) is compromised and subsequently uploads a (potentially even validly signed) tarball that differs from the public contents of the repository. I mentioned the binaries in general, but they have no direct relevance to Flathub, and noted that they are potentially far more prone to containing malicious elements than the tarball.

I wished to bring this up for consideration because I am not knowledgeable enough to assess which is better, since a password manager is a very sensitive application.

If the signed tarballs uploaded by a trusted maintainer is more secure and trustworthy than fetching the contents of the Github repository using Github's URLs, then please close this issue as invalid. In that case, it would be ideal if the signature of the tarball could be verified similarly to Arch prior to updating the Flatpak to newer versions.

**Comment by AsavarTzeth** [comment]
That makes sense, thank you. To be honest I don't feel knowledgeable enough either but I do have at least a basic understanding of trust and security.

The way I see it we trust that one core KeePassXC developer in any case. It also seems a bit silly not to trust someone that has been that project. If that trust is misplaced that project is already compromised.

I would actually argue that as a third party I am a much greater threat in this trust chain than any keepassxc core developer. This account could be compromised and for all you know so could I. Although everything that is built by Flathub is readable here something malicious could certainly get in, at the very least for a limited time. *Of course the same applies to practically every single distribution/package maintainer out there.*

To summarize: I don't consider your argument invalid, but I am doubtful there would be much, if any, security gain from this change. I will however leave this open for debate. Fair enough?

**Comment by imchau** [comment]
Seeing as major distributions like Arch and [Fedora](https://src.fedoraproject.org/rpms/keepassxc/blob/master/f/keepassxc.spec) fetch the maintainer-uploaded tarballs, which are also signed with PGP and have official checksums, I have come to the conclusion that Flathub should also stick with the same source. For all we know, Github could become compromised and serve malicious repository archives under those URLs.

I should have framed this as a question from that start, rather than as an issue that is potentially invalid. Thank you for taking the time to discuss this with me.

In case a compelling argument arises regarding this matter and sparks a discussion, this issue can be reopened. Otherwise it will just pollute the issue tracker.
